### PR TITLE
Fix combobox minimize

### PR DESCRIPTION
**Pull request by mollfpr**
### Details

Adding history router with type drawer to only when on mobile. On web or desktop size we don't need the drawer to toggle.

### Fixed Issues

$ https://github.com/Expensify/App/issues/8841

### Tests

1. Click on one chat first
2. Minimize the window as much as possible
3. Maximize the window
4. Click another chat
5. Minimize window
6. Verify that the report screen is showing when minimizing the window

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/docs/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [ ] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/docs/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Click on one chat first
2. Minimize the window as much as possible
3. Maximize the window
4. Click another chat
5. Minimize window
6. Verify that the report screen is showing when minimizing the window

- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/25520267/177384055-514a3fb7-d4d3-4ef1-b614-9951d5cdb93a.mp4

#### Mobile Web

https://user-images.githubusercontent.com/25520267/177385809-8c276067-d666-49e0-add8-31bba804957b.mp4

https://user-images.githubusercontent.com/25520267/177385870-316f8d8a-08dd-4950-a213-acf159a8ede6.mp4

#### Desktop

https://user-images.githubusercontent.com/25520267/177386129-b1ff6179-0296-4e9b-a2fe-6e60a56c1dac.mp4

#### iOS

https://user-images.githubusercontent.com/25520267/177600273-8eadde99-7836-4196-9aec-a2d14b313e69.mov

### CODE REVIEW

**Comment by mollfpr** [comment]
@Santhosh-Sellavel Can you help me to test on Android native? I can't run the android on my local. Thanks!

File: src/libs/Navigation/Navigation.js
Comment (aldo-expensify) [inline src/libs/Navigation/Navigation.js:82]: This getter having a side effect of setting `isSmallScreenWidth` feels too dirty. `navigate` calling `navigationRef.current.dispatch(CustomActions.pushDrawerRoute(route, isSmallScreenWidth));` depends on `getDefaultDrawerState` having been called before.

**Review by aldo-expensify — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/Navigation.js
Comment (mollfpr) [reply src/libs/Navigation/Navigation.js:82]: Are we okay using your approach to get the dimension?
```
const initialDimensions = Dimensions.get('window');
const isSmallScreenWidth = initialDimensions.width <= variables.mobileResponsiveWidthBreakpoint;
```

Comment (mollfpr) [reply src/libs/Navigation/Navigation.js:82]: I couldn't find another way to get the screen size except using `Dimensions`.

Comment (aldo-expensify) [reply src/libs/Navigation/Navigation.js:82]: This is what I came with when I was testing, I didn't see any other way that didn't involve changing too many things (I must say I didn't tried much :P)

Thoughts @Santhosh-Sellavel ?

Comment (Santhosh-Sellavel) [reply src/libs/Navigation/Navigation.js:82]: Isn't there any API in Dimensions to listen to the changes? @mollfpr Explore the options.

Comment (Santhosh-Sellavel) [reply src/libs/Navigation/Navigation.js:82]: > Are we okay using your approach to get the dimension?
> 
> ```
> const initialDimensions = Dimensions.get('window');
> const isSmallScreenWidth = initialDimensions.width <= variables.mobileResponsiveWidthBreakpoint;
> ```

This won't work.

Comment (aldo-expensify) [reply src/libs/Navigation/Navigation.js:82]: > This won't work.

Why?

Comment (Santhosh-Sellavel) [reply src/libs/Navigation/Navigation.js:82]: We measure only once here, then minimizing or expanding the window would not reflect correctly.

Comment (aldo-expensify) [reply src/libs/Navigation/Navigation.js:82]: I may be missing something, but if that code is inside `pushDrawerRoute`, wouldn't it be up to date when that function is called? `pushDrawerRoute` doesn't get called when minimizing or expanding the window, does it?

File: src/libs/Navigation/CustomActions.js
Comment (mollfpr) [inline src/libs/Navigation/CustomActions.js:66]: @Santhosh-Sellavel @aldo-expensify I adding a listener to get the screen size. It's should reflect the screen size.

Comment (aldo-expensify) [reply src/libs/Navigation/CustomActions.js:66]: hum, I still don't understand how `Dimensions.get('window');` would be outdated. From the documentation it looks like it should return the current width, up to date when is used by `pushDrawerRoute`.

I don't think this is necessary at all.

@Santhosh-Sellavel can you explain more why you think it would "not reflect correctly"?

File: src/libs/Navigation/Navigation.js
Comment (Santhosh-Sellavel) [reply src/libs/Navigation/Navigation.js]: > Yep, that should do.

@aldo-expensify I already confirmed it would work.

Comment (aldo-expensify) [reply src/libs/Navigation/Navigation.js]: > > Yep, that should do.
> 
> @aldo-expensify I already confirmed it would work.

The version with the listener?

Comment (Santhosh-Sellavel) [reply src/libs/Navigation/Navigation.js]: > Are we okay using your approach to get the dimension?
> 
> ```
> const initialDimensions = Dimensions.get('window');
> const isSmallScreenWidth = initialDimensions.width <= variables.mobileResponsiveWidthBreakpoint;
> ```

Sorry for the confusion, I thought we are determining once. This way itself will work if it is determined every time.
@aldo-expensify

File: src/libs/Navigation/CustomActions.js
Comment (Santhosh-Sellavel) [reply src/libs/Navigation/CustomActions.js:66]: So let's go with the first way right @aldo-expensify or let's keep this? I'll leave it to you to decide.

Comment (Santhosh-Sellavel) [reply src/libs/Navigation/CustomActions.js:66]: Cool! @mollfpr kindly update the PR

**Comment by mollfpr** [comment]
@aldo-expensify @Santhosh-Sellavel updated!

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel Can you help me to test on Android native? I can't run the android on my local. Thanks!

What issue you are facing in the local?

**Comment by Santhosh-Sellavel** [comment]
@mollfpr 
We have a breakage here

Steps:
1. Open the app in the expanded window.
2. Click any chat
3. Minimize the window
4. Click back from the chat header
5. Try to switch to a different chat but it's not working.

https://user-images.githubusercontent.com/85645967/179294812-4fd8eb7c-fb78-4c62-9ac5-2a5113097ef1.mov

cc: @aldo-expensify

**Comment by aldo-expensify** [comment]
@Santhosh-Sellavel I tried to reproduce the problem you see, but didn't happen to me. Does it always happen to you? or sometimes? can you check if you don't have the same problem in `main`?

**Comment by Santhosh-Sellavel** [comment]
I pulled from main and made the changes.

**Comment by mollfpr** [comment]
@Santhosh-Sellavel I'm unable to produce the above issue.

**Comment by Santhosh-Sellavel** [comment]
Reviewing again now.

**Review by aldo-expensify — APPROVED** [review]
Tested again and it seems to work fine.

I'll get another pair of eyes just to be sure.

**Review by Justicea83 — APPROVED** [review]
LGTM!

**Comment by Santhosh-Sellavel** [comment]
#### PR Reviewer Checklist</h4>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

**Review by Santhosh-Sellavel — APPROVED** [review]
Looks good to me!

**Comment by aldo-expensify** [comment]
thanks for reviewing again @Santhosh-Sellavel !

**Comment by marcaaron** [comment]
I do wish that a broader conversation would have been started about this PR and solution. Almost every issue dealing with `react-navigation` was put on hold a while back while so we could look at longer term more holistic solutions. This one still made it through somehow...

@Santhosh-Sellavel please recall these posts (maybe you have not seen them)

https://expensify.slack.com/archives/C02NK2DQWUX/p1652908236507109
https://expensify.slack.com/archives/C02NK2DQWUX/p1653333895007009

@aldo-expensify I'm not sure what I could have done differently here and not casting any blame for accepting this solution. Maybe I could have been more proactive here or there is some way to lock the file so it won't be edited anymore.

**Pull request merged**
